Automattic/wp-calypso
# Purchases: Reduxify Stored Cards on Secure Payment page

This pull request is a follow-up of https://github.com/Automattic/wp-calypso/pull/6613, https://github.com/Automattic/wp-calypso/pull/6628, and https://github.com/Automattic/wp-calypso/pull/6665 that updates the `Secure Payment` page in the checkout to use the global state tree to manage credit cards (stored during previous purchases or when a user added a new payment method to an existing purchase) instead of the [old singleton](https://github.com/Automattic/wp-calypso/tree/master/client/lib/stored-cards):

![screenshot](https://cloud.githubusercontent.com/assets/594356/16772722/9868f8c6-4855-11e6-899c-38f2d6e197a8.png)

#### Testing instructions
 
1. Run `git checkout update/reduxify-checkout` and start your server, or open a [live branch](https://calypso.live/?branch=update/reduxify-checkout)
2. Sandbox the store
3. Purchase e.g. a [premium theme](http://calypso.localhost:3000/design), a [plan](http://calypso.localhost:3000/plans), or a [site redirect](http://calypso.localhost:3000/domains/add/site-redirect) with a fake credit card
4. Open the [`Billing History` page](http://calypso.localhost:3000/me/billing)
5. Check that you can see the card with the right information in the `Manage Your Credit Cards` section
6. Add another purchase to your shopping cart and head to checkout
7. Check that you can see the previous credit card on the `Secure Payment` page
8. Enter a new fake credit card and click the `Pay` button
9. Repeat steps #4 to #5

#### Additional notes

This pull requests also updates the query component for stored cards to fetch data as soon as it is mounted, and only in that case. This is to avoid having to explicitly fetch stored cards when a user uses a new credit card during purchase or adds a new payment method to an existing purchase - as otherwise the `Billing History` page will display stale information if it was accessed by the user beforehand during a same session.

The logic here is that it indeed doesn't make sense to perform this call in these two use cases since there is little chance that the user visits the `Billing History` page afterwards. It seems to be more effective to fetch stored cards every time the user access this specific page, which I assume doesn't receive lots of traffic.


#### Reviews
 
 
@Automattic/sdev-feed
